### PR TITLE
Switch to CMake build for native binaries (xplat part 1)

### PR DESCRIPTION
**Pull request by swernli**
This change switches the build for the native LibLlvm binaries to be cmake based instead of msbuild and csproj based. This is the first step toward enabling cross-platform build the Llvm.NET as a whole (see #193).

### CODE REVIEW

**Comment by swernli** [comment]
@smaillet This change is ready for review!

File: src/Interop/LibLLVM/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,208 @@
+cmake_minimum_required(VERSION 3.0.0)
```
Comment (smaillet) [inline src/Interop/LibLLVM/CMakeLists.txt:1]: This is missing handling of rthe RC file, which as discussed in prior PRs is used to mark the version information in the file. A lot of effort went into ensuring that was done correctly and consistently for the build. Please restore that functionality.

Comment (smaillet) [reply src/Interop/LibLLVM/CMakeLists.txt:1]: See: https://stackoverflow.com/questions/8183156/cmake-and-visual-studio-resource-files for some suggestions on how to do that.

**Review by smaillet — CHANGES_REQUESTED** [review]
Thanks for the focused PR.
As mentioned in comments to the other PRs and this one, please add support for processing the RC files instead of just deleting them. In the future, please use the [Discussions section](https://github.com/UbiquityDotNET/Llvm.NET/discussions) of the GitHub project to discuss changes to the core code, build, etc... PRs without surprises save everyone troubles of being put into uncomfortable positions.

**Pull request closed without merging**